test/e2e/density.go

```diff
@@ -231,9 +235,38 @@ var _ = Describe("Density", func() {
 		}
 		itArg := testArg
 		It(name, func() {
+			uuid := string(util.NewUUID())
 			totalPods := itArg.podsPerMinion * minionCount
-			RCName = "my-hostname-density" + strconv.Itoa(totalPods) + "-" + string(util.NewUUID())
+			nameStr := strconv.Itoa(totalPods) + "-" + uuid
+			ns = "e2e-density" + nameStr
+			RCName = "my-hostname-density" + nameStr
 			RunRC(c, RCName, ns, "gcr.io/google_containers/pause:go", totalPods)
+			By("waiting for all events to be recorded")
+			last := -1
+			current := 0
+			var events *api.EventList
+			for last < current {
+				e, err := c.Events(ns).List(
+					labels.Everything(),
+					fields.Set{
+						"involvedObject.namespace": ns,
+					}.AsSelector(),
+				)
+				expectNoError(err)
+				last = current
```

Comment:
@lavalamp I'm not sure I see how using 6546 will help ensure that we inspect all the events that are logged.  I originally only cared about the start events, but there's no reason not to care about the shutdown events as well.  If I move the inspection of the events until after the rc is deleted I should be able to wait for all the events to be logged.  As long as the system is properly ensuring pods are shut down then the event stream should stop within a reasonable time after all the pods are stopped.  This should bound the test and prevent it from running forever.

I'm not sure how to do that with 6546 though.  I don't see any way to use that change and know that events have all been generated and it's ok to stop the controller.